com.unity.render-pipelines.high-definition/Editor/Wizard/HDProjectSettings.cs

```diff
@@ -34,9 +44,51 @@ enum Version
         bool m_WizardNeedRestartAfterChangingToDX12 = false;
         [SerializeField]
         bool m_WizardNeedToRunFixAllAgainAfterDomainReload = false;
+
+        // Note: this is for the *material version*, which uses MaterialPostProcessor k_Migrations[] as migration
+        // functions table. See note on Version enum above.
         [SerializeField]
         int m_LastMaterialVersion = k_NeverProcessedMaterialVersion;
 
+        [SerializeField]
+        ShaderGraphVersion m_HDShaderGraphLastSeenVersion = ShaderGraphVersion.NeverMigrated;
+
+        // This tracks latest versions of every plugin subtargets ever seen in the project.
+        // The dictionary key is the SubTarget GUID.
+        // If the project has ever seen an external-to-HDRP ("plugin") material, it saves the version
+        // of the material here. That way, we can use the same logic to forward upgrade calls
+        // than for the internally known HDRP materials: if a plugin material version is higher than
+        // what is seen here, we can trigger a re-import (see RegisterUpgraderReimport in MaterialPostProcessor).
+        [SerializeField]
+        PluginMaterialVersions m_PluginMaterialVersions = new PluginMaterialVersions();
+
+        // Same as above but for the *subtarget* versioning specific to the subtarget (ie not HDRP ShaderGraphVersion,
+        // see m_HDShaderGraphLastSeenVersion above for that).
+        [SerializeField]
+        PluginMaterialVersions m_PluginSubTargetVersions = new PluginMaterialVersions();
+
+        // The following filters m_PluginMaterialVersions on load/save based on plugin subtargets
+        // with the code base actually still present in the project.
+        // Unused for now as we can't filter materials per active shader type so we reimport all materials
+        // at the same time.
+        [NonSerialized]
+        Dictionary<GUID, (int, int)> m_InProjectPluginLastSeenMaterialAndSubTargetVersions = new Dictionary<GUID, (int, int)>();
+
+        /// <summary>
+        /// This is to quickly decide if some materials depending on plugin subtargets
+        /// (that have actually their plugin code present in the project) might need upgrade.
+        /// See pluginSubTargetMaterialVersionSum in HDShaderUtils: summed versions in here
+        /// should never be allowed to be higher than that returned by the currently present
+        /// (in project) plugin SubTarget codebase - ie that returned by
+        /// <by cref="IPluginSubTargetMaterialUtils"/>.
+        /// <See also cref="MaterialReimporter.RegisterUpgraderReimport"/>
+        /// </summary>
+        [NonSerialized]
+        long m_InProjectPluginLastSeenMaterialVersionsSum = (int)PluginMaterial.GenericVersions.NeverMigrated;
+
```

Comment:
Summing versions (caped by the codebase respective highest versions) for each plugin allows a single check to decide if we need to proceed to reimport all materials for migration in the polling function (in RegisterUpgraderReimport) regardless of the number of plugin materials present.